File: CS50/C/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int altura;

    do
    {
        altura = get_int("Altura: ");
    }
    while (altura < 1 || altura > 8);

    for (int linha = 1; linha <= altura; linha++)
    {
        // imprime espaços
        for (int espaco = 0; espaco < altura - linha; espaco++)
        {
            printf(" ");
        }

        // imprime #
        for (int hash = 0; hash < linha; hash++)
        {
            printf("#");
        }

        // pula linha
            printf("  ");
            for (int hash = 0; hash < linha; hash++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: CS50/Python/mario.py
from cs50 import get_int

def main():
    # Solicita a altura ao usuário
    while True:
        height = get_int("Height: ")
        if height >= 1 and height <= 8:
            break
    
    # Constrói a pirâmide
    for i in range(1, height + 1):
        # Calcula o número de espaços e hashes
        spaces = height - i
        hashes = i
        
        # Imprime a linha
        print(" " * spaces + "#" * hashes)

if __name__ == "__main__":
    main()
